server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -355,6 +356,11 @@ public TabletServerMinCMetrics getMinCMetrics() {
   private final ZooAuthenticationKeyWatcher authKeyWatcher;
   private final WalStateManager walMarker;
 
+  private int maxThreads =
+      getServerConfig().getConfiguration().getCount(Property.TSERV_MAX_WRITETHREADS);
+  private int maxThreadPermits = maxThreads == 0 ? Integer.MAX_VALUE : maxThreads;
```

Comment:
I would really like to be able to set this property dynamically (i.e. while the tservers are running).  Perhaps in the getSemaphore method we could periodically (once every 5 seconds or so) check the TSERV_MAX_WRITETHREADS property and then recreate the Semaphore with the new max and copy in the current value.  This could be a separate ticket however.

Reply:
One way to do what I suggest is to rewrite the getSemaphore method as follows:
```
  private Semaphore getSemaphore() {
    int writeThreads = getServerConfig().getConfiguration().getCount(Property.TSERV_MAX_WRITETHREADS);
    if (writeThreads == 0) writeThreads = Integer.MAX_VALUE;
    if (sem == null || maxThreadPermits != writeThreads) {
      synchronized (this) {
        maxThreadPermits = writeThreads;
        sem = new Semaphore(maxThreadPermits);
      }
    }
    return sem;
  }
```

Then down below where you get the semaphore, preserve the semaphore that you use for the tryAcquire and use the same one for the release.

Reply:
The semaphoreAcquired boolean could be replaced with the Semaphore instance instead, then instead of "if (semaphoreAcquired)" it would be "if (semaphore != null)".

Reply:
my only problem with this is the getServerConfig().getConfiguration().getCount(property) may be too expensive to do every time.  Adding a counter or a time mechanism may be required there.

Reply:
Alternatively we add a timer that will check the property every N seconds and update the semaphore as required.

Reply:
if it is not too expensive then this simple solution may suffice.